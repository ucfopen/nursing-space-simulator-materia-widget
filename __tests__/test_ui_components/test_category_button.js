import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import CategoryButton from "../../src/js/components/ui/category_button";

// Constants used across all tests
const mockClick = jest.fn();

describe("Category button tests", () => {
	it("renders correctly", () => {
		const categoryButtonTree = renderer
			.create(<CategoryButton category="beds" onClick={mockClick} />)
			.toJSON();

		expect(categoryButtonTree).toMatchSnapshot();
	});

	it("handles click correctly", () => {
		const categoryButton = shallow(
			<CategoryButton category="beds" onClick={mockClick} />
		);

		categoryButton.find("button").simulate("click");
		expect(mockClick).toBeCalled();
	});
});
