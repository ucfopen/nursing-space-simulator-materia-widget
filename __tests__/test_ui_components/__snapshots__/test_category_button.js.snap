// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category button tests renders correctly 1`] = `
<button
  className="asset-category"
  data-category="beds"
  onClick={[Function]}
>
  beds
</button>
`;
